lib/geomancer/geo_json.ex

```diff
@@ -15,9 +17,88 @@ defmodule Geomancer.GeoJson do
             name: nil,
             bbox: []
 
-  @spec new(features :: [Feature.t()], name :: String.t(), bbox :: map()) :: t()
-  def new(features, name, bbox) do
-    bounding = [bbox.xmin, bbox.ymin, bbox.xmax, bbox.ymax]
-    %__MODULE__{features: features, name: name, bbox: bounding}
+  @spec new({:ok, struct()} | {:error, String.t()}) :: t() | {:error, String.t()}
+  def new({:ok, %Geomancer.Shapefile{} = shapefile}) do
```

Comment:
I feel real weird about having an :ok tuple as an argument. I'd feel better about it if this were a private method. Otherwise I'd rather see it handled with a "railroad with" in the from but maybe I'm just being sorta pedantic.

Reply:
🤔